packages/manager/apps/pci-databases-analytics/src/pages/services/[serviceId]/users/_components/formUser/AclsSelect.component.tsx

```
@@ -0,0 +1,212 @@
+    );
+    const patternInputRef = useRef<HTMLInputElement>(null);
+    const addAclBtnRef = useRef<HTMLButtonElement>(null);
+    const scrollListRef = useRef<HTMLUListElement>(null);
+    const { form, schema } = useAclsSelectForm({
+      existingAcls: value,
+    });
+    type ValidationSchema = z.infer<typeof schema>;
+
+    // Scroll to the bottom of the list when a new acl is added
+    useEffect(() => {
+      const scrollContainer = scrollListRef.current;
+      if (scrollContainer) {
+        const { lastChild } = scrollContainer;
+        if (lastChild instanceof Element) {
+          lastChild.scrollIntoView({ behavior: 'smooth' });
+        } else {
+          scrollContainer.scrollTop = scrollContainer.scrollHeight;
+        }
+      }
+    }, [value, scrollListRef]);
+
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if (event.key === 'Enter') {
+        addAclBtnRef.current?.click();
+        patternInputRef.current?.focus();
+        event.preventDefault();
+      }
+    };
+    const handleAddAcl: SubmitHandler<ValidationSchema> = (formValues) => {
+      onChange([
+        ...value,
+        {
+          pattern: formValues.pattern,
+          permission: formValues.permission,
+        },
+      ]);
+      form.reset();
+      patternInputRef.current?.focus();
+    };
+    const handleRemoveAcl = (index: number) => {
+      const updatedAcls = [...value];
+      updatedAcls.splice(index, 1);
+      onChange(updatedAcls);
+    };
+
+    const errors = useMemo(() => {
+      const messages: string[] = [];
+      const formErrors = form.formState.errors;
+      Object.keys(formErrors).forEach((key) => {
+        const validKey = key as keyof typeof formErrors;
+        const formError = formErrors[validKey];
+        if (formError?.message) {
+          messages.push(formError.message);
+        }
+      });
+
+      return messages;
+    }, [form.formState.errors]);
```

IMO, type cast is pretty weak because if `key` is not part of `formErrors` it could raise a silent error .
BTW, it could be simplified using filter + map to prevent from `push` mutability :)